tests/Unit/ParallelAlgorithms/Events/Test_ObserveVolumeIntegrals.cpp

```diff
@@ -0,0 +1,261 @@
+  fill_with_random_values(make_not_null(&vars), make_not_null(&gen),
+                          make_not_null(&dist));
+
+  // Compute expected data for checks
+  size_t num_tensor_components = 0;
+  const DataVector det_jacobian =
+      get(determinant(map.jacobian(logical_coords)));
+  const double expected_volume = definite_integral(det_jacobian, mesh);
+  std::vector<double> expected_volume_integrals{};
+  std::vector<std::string> expected_reduction_names = {
+      db::tag_name<ObservationTimeTag>(), "Volume"};
+  tmpl::for_each<typename std::decay_t<decltype(vars)>::tags_list>([
+    &num_tensor_components, &vars, &expected_volume_integrals,
+    &expected_reduction_names, &det_jacobian, &mesh
+  ](auto tag) noexcept {
+    using tensor_tag = tmpl::type_from<decltype(tag)>;
+    const auto tensor = get<tensor_tag>(vars);
+    for (size_t i = 0; i < tensor.size(); ++i) {
+      expected_reduction_names.push_back("VolumeIntegral(" +
+                                         db::tag_name<tensor_tag>() +
+                                         tensor.component_suffix(i) + ")");
+      expected_volume_integrals.push_back(
+          definite_integral(det_jacobian * tensor[i], mesh));
+      ++num_tensor_components;
+    }
+  });
+
+  const double observation_time = 2.0;
+  const auto box = db::create<
+      db::AddSimpleTags<ObservationTimeTag, domain::Tags::Mesh<volume_dim>,
+                        domain::Tags::ElementMap<volume_dim, Frame::Inertial>,
+                        domain::Tags::Coordinates<volume_dim, Frame::Logical>,
+                        Tags::Variables<typename decltype(vars)::tags_list>>>(
+      observation_time, std::move(mesh), std::move(map),
+      std::move(logical_coords), std::move(vars));
+
+  ActionTesting::MockRuntimeSystem<metavariables> runner{{}};
+  ActionTesting::emplace_component<element_component>(make_not_null(&runner),
+                                                      0);
+  ActionTesting::emplace_component<observer_component>(&runner, 0);
+
+  observe->run(box, runner.cache(), array_index,
+               std::add_pointer_t<element_component>{});
+
+  // Process the data
+  runner.template invoke_queued_simple_action<observer_component>(0);
+  CHECK(runner.template is_simple_action_queue_empty<observer_component>(0));
+
+  const auto& results = MockContributeReductionData::results;
+  CHECK(results.observation_id.value() == observation_time);
+  CHECK(results.subfile_name == "/volume_integrals");
+  CHECK(results.reduction_names == expected_reduction_names);
+  CHECK(results.time == observation_time);
+  CHECK(results.volume == expected_volume);
+  CHECK(results.volume_integrals.size() == num_tensor_components);
+  CHECK(results.volume_integrals == expected_volume_integrals);
+}
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Evolution.dG.ObserveVolumeIntegrals",
+                  "[Unit][Evolution]") {
+  using system = SomeSystem<3>;
+
+  INFO(pretty_type::get_name<system>());
```

Comment:
Add scopes around these "sections" in your test so that the `INFO` only gets printed if a check within this section fails:

```cpp
{
  INFO(...);
  // check things
}
```